app/action-endpoint.ts

```diff
@@ -55,9 +63,35 @@ export const handler = apiWrapper(async ({ body, success, error }: ApiSignature)
             await changeMembership(initiativeId, teamId, slackUserId, false);
             break;
           }
+          case MemberAction.OPEN_EDIT_DIALOG: {
+            dialogResponse = true;
+            response = new EditInitiativeDialogResponse(initiative, triggerId);
+            break;
+          }
+        }
+        if (action !== MemberAction.OPEN_EDIT_DIALOG) {
+          response = new DetailResponse(initiative, slackUserId, channel);
+        }
+        break;
+      }
+      case InitiativeCallbackAction.EDIT_INITIATIVE_DIALOG: {
+        const slackUserId = payload.user.id;
+        const { initiative_name, initiative_description } = payload.submission;
+        const { originalName, originalDescription, initiativeId } = JSON.parse(payload.state);
+        const fieldValidator = new EditInitiativeFieldValidator(
```

Comment:
can we move some of this validator logic into a helper method in this file?  so that here in the switch statement it calls that method and gets the correct response, and all the logic for determining the correct response lives in that method